src/common/libsubprocess/subprocess.h

```diff
@@ -215,6 +215,11 @@ const char *flux_subprocess_read (flux_subprocess_t *p,
  *   be NUL terminated.  If no line is available, returns pointer and
  *   length of zero.  User shall not free returned pointer.  Length of
  *   buffer returned can optionally returned in 'lenp'.
+ *
+ *   Use flux_subprocess_read_stream_closed() to determine if stream
+ *   is closed, as length of zero cannot be assumed that subprocess
+ *   has closed the stream (i.e. data is buffered, but not a line
+ *   worth of data).
```

Comment:
I'm finding "as length of zero cannot be assumed that subprocess has closed the stream"   a little hard on the ears :-)

Could this paragraph just be replaced with

> A length of zero may be returned if the stream is closed OR if the stream is line buffered and a line is not yet available.  Use `flux_subprocess_read_stream_closed()` to distinguish between the two.